cassandra-analytics-spark-converter/src/main/java/org/apache/cassandra/spark/data/converter/types/complex/SparkMap.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cassandra.spark.data.converter.types.complex;
+
+import java.util.HashMap;
+import java.util.stream.Collectors;
+
+import org.apache.cassandra.bridge.BigNumberConfig;
+import org.apache.cassandra.spark.data.CqlField;
+import org.apache.cassandra.spark.data.converter.SparkSqlTypeConverter;
+import org.apache.cassandra.spark.data.converter.types.SparkType;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.expressions.GenericInternalRow;
+import org.apache.spark.sql.catalyst.util.ArrayBasedMapData;
+import org.apache.spark.sql.catalyst.util.ArrayData;
+import org.apache.spark.sql.catalyst.util.MapData;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.jetbrains.annotations.NotNull;
+import scala.collection.JavaConverters;
+
+public class SparkMap implements MapFeatures
+{
+    private final SparkSqlTypeConverter converter;
+    private final CqlField.CqlMap map;
+
+    public SparkMap(SparkSqlTypeConverter converter, CqlField.CqlMap map)
+    {
+        this.converter = converter;
+        this.map = map;
+    }
+
+    public DataType dataType(BigNumberConfig bigNumberConfig)
+    {
+        return DataTypes.createMapType(keyType().dataType(bigNumberConfig),
+                                       valueType().dataType(bigNumberConfig));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Object toSparkSqlType(@NotNull Object value, boolean isFrozen)
+    {
+        return mapToSparkSqlType((java.util.Map<Object, Object>) value, isFrozen);
+    }
+
+    private ArrayBasedMapData mapToSparkSqlType(final java.util.Map<Object, Object> map, boolean isFrozen)
+    {
+        final Object[] keys = new Object[map.size()];
+        final Object[] values = new Object[map.size()];
+        int position = 0;
+        for (final java.util.Map.Entry<Object, Object> entry : map.entrySet())
```

Comment:
NIT Remove final qualifiers
```suggestion
        Object[] keys = new Object[map.size()];
        Object[] values = new Object[map.size()];
        int position = 0;
        for (java.util.Map.Entry<Object, Object> entry : map.entrySet())
```